src/include/binder/binder_sherpa.h

```diff
@@ -31,41 +32,32 @@ class BinderSherpa {
    * @param parse_result The parse result to be tracked
    * @param parameters parameters for the query being bound, can be nullptr if there are no parameters
    */
-  explicit BinderSherpa(const common::ManagedPointer<parser::ParseResult> parse_result,
-                        const common::ManagedPointer<std::vector<type::TransientValue>> parameters)
-      : parse_result_(parse_result), parameters_(parameters) {
-    TERRIER_ASSERT(parse_result != nullptr, "We shouldn't be tring to bind something without a ParseResult.");
-  }
+  explicit BinderSherpa(common::ManagedPointer<parser::ParseResult> parse_result,
+                        common::ManagedPointer<std::vector<type::TransientValue>> parameters);
 
   /**
    * @return The parse result that we're tracking.
    */
-  common::ManagedPointer<parser::ParseResult> GetParseResult() const { return parse_result_; }
+  common::ManagedPointer<parser::ParseResult> GetParseResult() const;
 
   /**
    * @return parameters for the query being bound
    * @warning can be nullptr if there are no parameters
    */
-  common::ManagedPointer<std::vector<type::TransientValue>> GetParameters() const { return parameters_; }
+  common::ManagedPointer<std::vector<type::TransientValue>> GetParameters() const;
```

Comment:
If I understand gcc correctly, the compiler cannot inline these functions if you put them in the cpp file. Just curious is there a reason to move these one-liner functions into the cpp file?

Reply:
You're right. For the one-liners I should just leave them in the headers.